### DISCUSSION TITLE
docker MIMIR cluster member IP does not support node IP

### DISCUSSION THREAD

**Discussion by CHASII** [grafana/mimir#6444] [Help and support]
#### Describe the bug
I attempted to use the vmware virtualization tool and virtualized three vm hosts with IP addresses of 1.1.1.8, 1.1.1.9, and 1.1.1.10. Then, I installed dockers on each of these three nodes and deployed mimir on these three hosts to form a cluster. But I found that no matter how configured, the host IP of mimir is the internal IP of 172. x.x.x, rather than 1.1.1. x.

#### To Reproduce

1. Start Mimir (SHA or version): Latest version

2. Perform operations (read/write/other)

Node 1:

Docker run - d -- name mimir1-- host name mimir1-- network host - v/data/mimir configuration/mimir. yaml:/etc/mimir. yaml - v/data/mimir data:/data grafana/mimir: latest configuration. file=/etc/mimir. yaml

Node 2:

Docker run - d -- name mimir2-- host name mimir2-- network host - v/data/mimir configuration/mimir. yaml:/etc/mimir. yaml - v/data/mimir data:/data grafana/mimir: latest configuration. file=/etc/mimir. yaml

Node Three:

Docker run - d -- name mimir3-- host name mimir3-- network host - v/data/mimir configuration/mimir. yaml:/etc/mimir. yaml - v/data/mimir data:/data grafana/mimir: latest configuration. file=/etc/mimir. yaml

#### Expected behavior

Check the logs of the Docker container and report the following error.

ts=2023-10-19T11:01:13.652148432Z caller=log.go:245 level=warn msg="Got ping for unexpected node 'mimir2-2c0dd7d0' from=172.17.0.1:7946"
ts=2023-10-19T11:01:23.655902433Z caller=log.go:245 level=warn msg="Got ping for unexpected node 'mimir2-2c0dd7d0' from=172.17.0.1:7946"
ts=2023-10-19T11:01:38.655180783Z caller=log.go:245 level=warn msg="Got ping for unexpected node 'mimir2-2c0dd7d0' from=172.17.0.1:7946"
ts=2023-10-19T11:01:58.652714104Z caller=log.go:245 level=warn msg="Got ping for unexpected node 'mimir3-a2d292e0' from=172.17.0.1:7946"
ts=2023-10-19T11:02:18.656545544Z caller=log.go:245 level=warn msg="Got ping for unexpected node 'mimir3-a2d292e0' from=172.17.0.1:7946"

#### Environment

docker version：newest
os versiont：centos7

mimir. yaml：from /docs/sources/mimir/get-started/play-with-grafana-mimir/config/mimir.yaml

#### Additional Context

My request is to form a mimir cluster through my host IP (1.1.1. x). Can I modify which configurations or environment variables to complete?

<img width="846" alt="微信图片_20231019225918" src="https://github.com/grafana/mimir/assets/12842732/e4806441-e44b-4ddb-ba1b-b3350f97bb2f">

<img width="863" alt="微信图片_20231019225925" src="https://github.com/grafana/mimir/assets/12842732/1abe70e8-8151-4616-84a1-0510a718bfad">

**Comment by charleskorn** [comment]
If the auto-detected IP address is not the address you'd like to use, you can manually set the desired address with the `instance_addr` configuration options for each component.

**Reply by CHASII** [reply]
thank you your reply，configuration options in mimir.yaml like this ：instance_addr： 1.1.1.8 ？ please

**Reply by charleskorn** [reply]
Please take a look at the [docs](https://grafana.com/docs/mimir/latest/references/configuration-parameters/) for configuration options - you'll need to set `instance_addr` in multiple places, once for each component that takes a `instance_addr` option.

**Reply by CHASII** [reply]
thank you~
